core/sql/sqlcomp/PrivMgrMD.cpp

```diff
@@ -630,6 +630,138 @@ bool PrivMgrMDAdmin::isAuthorized (void)
 }
 
 // ****************************************************************************
+// method:  getColumnReferences
+//
+//  This method stores the list of columns for the object in the 
+//  ObjectReference. 
+// **************************************************************************** 
+PrivStatus PrivMgrMDAdmin::getColumnReferences (ObjectReference *objectRef)
+{
+  std::string objectMDTable = trafMetadataLocation_ + ".OBJECTS o";
+  std::string colMDTable = trafMetadataLocation_ + ".COLUMNS c";
+
+  // Select column details for object 
+  std::string selectStmt = "select c.column_number from ";
+  selectStmt += objectMDTable;
+  selectStmt += ", ";
+  selectStmt += colMDTable;
+  selectStmt += " where o.object_uid = ";
+  selectStmt += UIDToString(objectRef->objectUID);
+  selectStmt += " and o.object_uid = c.object_uid";
+  selectStmt += " order by column_number";
```

Comment:
Why is the join to the OBJECTS table needed? Looks like the only thing you are retrieving is from the COLUMNS table and you have the key info you need for that?